tests/unit/classifier/svm/LibLinear_unittest.cc

```diff
@@ -6,1292 +6,469 @@
  *
  * Written (W) 2014 pl8787
  */
+#include <gtest/gtest.h>
+#include "utils/Utils.h"
 
 #include <shogun/classifier/svm/LibLinear.h>
 #include <shogun/features/DataGenerator.h>
 #include <shogun/features/DenseFeatures.h>
 #include <shogun/evaluation/ContingencyTableEvaluation.h>
 #include <shogun/mathematics/Math.h>
 #include <gtest/gtest.h>
+#include <map>
+#include <string>
+#include <vector>
 
 using namespace shogun;
 
 #ifdef HAVE_LAPACK
 
-//Generate Data for L1 regularized,
-//it's data need to be transposed first
-void generate_data_l1(CDenseFeatures<float64_t>* &train_feats,
-					 CDenseFeatures<float64_t>* &test_feats,
-					 CBinaryLabels* &ground_truth)
+class LibLinear : public ::testing::Test
 {
-	index_t num_samples = 50;
-	CMath::init_random(5);
-	SGMatrix<float64_t> data =
-		CDataGenerator::generate_gaussians(num_samples, 2, 2);
-	CDenseFeatures<float64_t> features(data);
-
-	SGVector<index_t> train_idx(num_samples), test_idx(num_samples);
-	SGVector<float64_t> labels(num_samples);
-	for (index_t i = 0, j = 0; i < data.num_cols; ++i)
-	{
-		if (i % 2 == 0)
-			train_idx[j] = i;
-		else
-			test_idx[j++] = i;
+public:
+	CDenseFeatures<float64_t>* train_feats;
+	CDenseFeatures<float64_t>* test_feats;
+	CBinaryLabels* ground_truth;
 
-		labels[i/2] = (i < data.num_cols/2) ? 1.0 : -1.0;
+	virtual void SetUp()
+	{
+		sg_rand->set_seed(1);
+	}
+	virtual void TearDown()
+	{
+		SG_UNREF(train_feats);
+		SG_UNREF(test_feats);
+		SG_UNREF(ground_truth);
 	}
+	//Helper that tests can call to drastically reduce code
+	void train_with_solver
+	(LIBLINEAR_SOLVER_TYPE llst, bool biasEnable, bool l1)
+	{
+		LIBLINEAR_SOLVER_TYPE liblinear_solver_type = llst;
 
-	train_feats = (CDenseFeatures<float64_t>*)features.copy_subset(train_idx);
-	test_feats =  (CDenseFeatures<float64_t>*)features.copy_subset(test_idx);
+		if (l1)
+				generate_data_l1();
+		else
+				generate_data_l2();
 
-	SGMatrix<float64_t> train_matrix = train_feats->get_feature_matrix();
-	SGMatrix<float64_t>::transpose_matrix(train_matrix.matrix,
-			train_matrix.num_rows, train_matrix.num_cols);
+		CLibLinear* ll = new CLibLinear();
```

Comment:
you can use some in many of these cases, so you dont need to call SG_UNREF :)
for example:
```auto ll = some<CLibLinear>();```

Reply:
Done